src/modules/global/components/RoutePrivateGuard.tsx

```diff
@@ -1,12 +1,34 @@
-import React, { PropsWithChildren } from "react";
+import React, { PropsWithChildren, useState } from "react";
 import { Navigate } from "react-router-dom";
-import { useRecoilValue } from "recoil";
-import { loggedState } from "../../authentification/state";
+import { useRecoilState, useSetRecoilState } from "recoil";
+import { loggedState, userState } from "../../authentification/state";
+import { auth } from "../../firebase";
+import { onAuthStateChanged } from "firebase/auth";
+import { PageLoading } from "./PageLoading";
 
 export const RoutePrivateGuard: React.FC<PropsWithChildren> = ({
   children,
 }) => {
-  const isLoggedIn = useRecoilValue(loggedState);
+  const [isLoggedIn, setIsLoggedIn] = useRecoilState(loggedState);
+  const setUserData = useSetRecoilState(userState);
+  const [isPageLoading, setIsPageLoading] = useState(true);
+
+  onAuthStateChanged(auth, (user) => {
+    if (!user) throw new Error("User could not be found.");
+    if (!user.displayName || !user.email)
+      throw new Error("User data coult not be found");
+    setIsPageLoading(false);
+
+    setUserData({
+      id: user.uid,
+      name: user.displayName,
+      email: user.email,
+    });
+    setIsLoggedIn(true);
+    return;
+  });
```

Comment:
This should be put inside `useEffect`, and you should cleanup the listener

Reply:
Put inside useEffect and returned the function.